install/kubernetes/cilium/values.yaml

```diff
@@ -588,6 +588,13 @@ eni:
   ec2APIEndpoint: ""
   # -- Tags to apply to the newly created ENIs
   eniTags: {}
+  # -- Interval for garbage collection of unattached ENIs. Set to "0s" to disable.
+  # @default -- `"5m"`
+  gcInterval: ""
+  # -- Additional tags attached to ENIs created by Cilium.
+  # Dangling ENIs with this tag will be garbage collected
+  # @default -- `{"io.cilium/cilium-managed":"true}`
```

Comment:
Is there ever a chance that we could accidentally select ENIs owned by a different cluster? Should we tag this with a cluster name?

Reply:
Yes, we will very likely collect ENIs owned by different clusters. I don't consider this a huge problem (the mechanism is explicitly designed to not need synchronization). But if users want/need to turn off GC for a particular cluster, they will have to either use per-cluster tags here, or set a global subnet tag filter unique to each cluster.

Unfortunately, I expect most users to not have a clustername set. `cilium install` does set one automatically, but most advanced users will use Helm and thus will have an empty clustername. But maybe we should still automatically add a cluster tag here _if_ the clustername is set and recommend it in the docs.

Reply:
Given that this only collects detached ENIs (correct me if I'm wrong), I assume it's not a big deal if we GC ENIs not "owned" by us, right? If so, this discussion is mostly academic.

But could we theoretically step on an ENI in the creation process?

Reply:
@squeed I believe this is handled by the GC logic waiting for the next cycle to GC ENIs. Basically, it's like a mark and sweep GC. First round we collect all the ENIs (mark) and the next round we delete (sweep). Please correct me if I'm wrong @gandro.

Reply:
@christarazi Yes, that's correct. Given an offline discussion with @squeed I will try to add a best-effort clustername tag as well, we might even be able to extract the EKS clustername from the EC2 API where available.

While generally speaking I don't think it is a problem if we GC ENIs from other clusters, if GC misbehaves and the user wants to turn it off or reconfigure, then cross-cluster GC will become an issue (since the settings in a different cluster will affect the GC in the local cluster and vice-versa).

Reply:
@squeed @christarazi I have updated the PR to auto-detect the clustername and tag ENIs with the clustername if available. Please take a look.